qiskit/dagcircuit/dagcircuit.py

```diff
@@ -566,22 +566,30 @@ def _map_condition(wire_map, condition):
         else:
             # if there is a condition, map the condition bits to the
             # composed cregs based on the wire_map
-            creg = condition[0]
+            cond_creg = condition[0]
             cond_val = condition[1]
             new_cond_val = 0
             new_creg = None
             for bit in wire_map:
-                if bit in creg:
+                if bit in cond_creg:
+                    try:
+                        candidate_creg = next(creg
+                                              for creg in target_cregs
+                                              if wire_map[bit] in creg)
+                    except StopIteration:
+                        raise DAGCircuitError('Did not find creg containing '
```

Comment:
It might be worth updating the docstring to add this error case under the `DAGCircuitError` raise section. Something like, "also when the specified condition is not present in a classical register"

Reply:
Added in 686ed2c058 .